livy/src/main/java/org/apache/zeppelin/livy/LivySparkSQLInterpreter.java

```diff
@@ -27,65 +28,73 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
 
 
 /**
- * Livy PySpark interpreter for Zeppelin.
+ * Livy SparkSQL Interpreter for Zeppelin.
  */
-public class LivySparkSQLInterpreter extends Interpreter {
+public class LivySparkSQLInterpreter extends BaseLivyInterprereter {
 
-  Logger LOGGER = LoggerFactory.getLogger(LivySparkSQLInterpreter.class);
+  private LivySparkInterpreter sparkInterpreter;
 
-  protected Map<String, Integer> userSessionMap;
-  private LivyHelper livyHelper;
+  private boolean sqlContextCreated = false;
 
   public LivySparkSQLInterpreter(Properties property) {
     super(property);
-    livyHelper = new LivyHelper(property);
-    userSessionMap = LivySparkInterpreter.getUserSessionMap();
   }
 
   @Override
-  public void open() {
+  public String getSessionKind() {
+    return "spark";
   }
 
   @Override
-  public void close() {
-    livyHelper.closeSession(userSessionMap);
+  public void open() {
+    super.open();
+    this.sparkInterpreter =
+        (LivySparkInterpreter) getInterpreterInTheSameSessionByClassName(
+            LivySparkInterpreter.class.getName());
   }
 
   @Override
-  public InterpreterResult interpret(String line, InterpreterContext interpreterContext) {
+  public InterpreterResult interpret(String line, InterpreterContext context) {
     try {
-      if (userSessionMap.get(interpreterContext.getAuthenticationInfo().getUser()) == null) {
-        try {
-          userSessionMap.put(
-              interpreterContext.getAuthenticationInfo().getUser(),
-              livyHelper.createSession(
-                  interpreterContext,
-                  "spark")
-          );
-        } catch (Exception e) {
-          LOGGER.error("Exception in LivySparkSQLInterpreter while interpret ", e);
-          return new InterpreterResult(InterpreterResult.Code.ERROR, e.getMessage());
-        }
+      if (StringUtils.isEmpty(line)) {
+        return new InterpreterResult(InterpreterResult.Code.SUCCESS, "");
       }
 
-      if (line == null || line.trim().length() == 0) {
-        return new InterpreterResult(InterpreterResult.Code.SUCCESS, "");
+      // create sqlContext implicitly if it is not available, as in livy 0.2 sqlContext
+      // is not available.
+      synchronized (this) {
+        if (!sqlContextCreated) {
+          InterpreterResult result = sparkInterpreter.interpret("sqlContext", context);
+          if (result.code() == InterpreterResult.Code.ERROR) {
+            result = sparkInterpreter.interpret(
+                "val sqlContext = new org.apache.spark.sql.SQLContext(sc)\n"
```

Comment:
It is just to make the behavior consistent with `SparkInterpreter` where SQLContext is always available no matter whether it is spark1 or spark2. But you are right, it is better to use SparkSession instead for spark2.